src/smtsolvers/CoreSMTSolver.cc

```diff
@@ -608,6 +602,23 @@ Lit CoreSMTSolver::pickBranchLit()
 
 }
 
+uint32_t CoreSMTSolver::calc_glue(const vec<Lit>& ps)
+{
+    MYFLAG++;
+    uint32_t nblevels = 0;
+    for (Lit lit: ps) {
+        int l = vardata[var(lit)].level;
+        if (l != 0 && permDiff[l] != MYFLAG) {
+            permDiff[l] = MYFLAG;
```

Comment:
```suggestion
        if (l != 0 && permDiff[lit] != MYFLAG) {
            permDiff[lit] = MYFLAG;
```

Reply:
@msooseth I was looking at the code finally now that I managed to run the experiments, and started wondering about these lines.  I'm pretty sure that `permDiff` should be indexed by lit and not the level, but I still wanted to check with you that I got this right.

Reply:
This is correct as-is, actually :) It's supposed to calculate the number of variables with different levels in the clause. Let's say you have a clause: `v1 v2 -v3 v4 v5`, and:
```
lev[v1] = 5
lev[v2] = 8
lev[v3] = 5
lev[v4] = 4
lev[v5] = 4
```

Then this function is supposed to return 3, because there are 3 distinct levels: 5, 8, and 4. What this function does is that it will take a new `MYFLAG`, say 99, which it never used before, and set `glue=0`. It will check `v1`'s level, so if `permDiff[5]` is 99, which it is not, so it sets `permDiff[5]=99`, and sets `glue++`. Then it checks `v2`'s permDiff, `permdiff[8]`, which is not 99, so it sets it to 99, and `glue++`. Now `v3`'s `permDiff[5]` has already been set to 99, so it doesn't increase `glue`. etc.

The `MYFLAG` will be new every time, and it's `uint64_t` so it will never overflow.

I hope this explains it! The paper is here:  https://www.ijcai.org/Proceedings/09/Papers/074.pdf

Reply:
Yes it does, thanks!